commit da444239df8742c8ecf4dc75ce36f9cd28021481
Author: Robert Mustacchi <rm@joyent.com>
Date:   2019-08-19T17:45:09+00:00 (7 weeks ago)
    
    OS-7950 Update gcc7 to gcc-7.3.0-il3
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Reviewed by: John Levon <john.levon@joyent.com>
    Approved by: John Levon <john.levon@joyent.com>

diff --git a/gcc7/Makefile b/gcc7/Makefile
index 0c634aa..889bce8 100644
--- a/gcc7/Makefile
+++ b/gcc7/Makefile
@@ -10,20 +10,23 @@
 #
 
 #
-# Copyright (c) 2018, Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 #
-# The GCC tarball is a copy of the https://github.com/illumos/gcc repository,
-# gcc-7.3.0-il-1 branch.  This includes all of the Illumos-specific
-# modifications we need.  We could probably do a git clone instead, but the
-# tarball method is easier given these Makefiles.
+# The GCC tarball is the gcc-7.3.0-il-3 release tag of the
+# https://github.com/illumos/gcc repository.  This includes all of the
+# illumos-specific modifications we need.  We could probably do a git
+# clone instead, but the tarball method is easier given these Makefiles.
+# The reason that there are two gcc entiries below is because the first
+# is used by github to indicate the repository and the second is part of
+# the tag.
 #
 
 include ../Makefile.defs
 
 GCC_VER = 7
-VER = gcc-7.3.0-il-1
+VER = gcc-gcc-7.3.0-il-3
 MPFR_VER = mpfr-3.1.6
 GMP_VER = gmp-6.1.2
 MPC_VER = mpc-1.0.3
diff --git a/gcc7/Patches/1001-unwind-map.patch b/gcc7/Patches/1001-unwind-map.patch
deleted file mode 100644
index 39e8048..0000000
--- a/gcc7/Patches/1001-unwind-map.patch
+++ /dev/null
@@ -1,51 +0,0 @@
-From d4c95f6d7ddff763faf4dcafcac0d083d9e5b801 Mon Sep 17 00:00:00 2001
-From: John Levon <john.levon@joyent.com>
-Date: Fri, 16 Nov 2018 10:13:59 +0000
-Subject: [PATCH] libgcc-unwind.map should be version 2 for SunOS ld, to
- prevent -z guidance from complaining.
-
----
-diff --git a/libgcc/config/t-slibgcc-sld b/libgcc/config/t-slibgcc-sld
-index 0b95391..55fd6e1 100644
---- a/libgcc/config/t-slibgcc-sld
-+++ b/libgcc/config/t-slibgcc-sld
-@@ -8,13 +8,32 @@ ifeq ($(enable_shared),yes)
- 
- # Linker mapfile to enforce direct binding to libgcc_s unwinder
- # (PR target/59788).
-+
-+ifeq ($(with_gnu_ld),yes)
-+
-+libgcc-unwind.map: libgcc-std.ver
-+	@(echo "{";                             \
-+	for f in `grep _Unwind_ $< | sort`; do  \
-+	  echo "        $$f = EXTERN DIRECT;";  \
-+	done;                                   \
-+	echo "};" ) > $@
-+
-+else
-+
- libgcc-unwind.map: libgcc-std.ver
--	@(echo "{";				\
--	for f in `grep _Unwind_ $< | sort`; do	\
--	  echo "	$$f = EXTERN DIRECT;";	\
--	done;					\
-+	@(echo "# Generated by libgcc/config/t-slibgcc-sld";	\
-+	echo;							\
-+	echo '$$mapfile_version 2'; 				\
-+	echo;							\
-+	echo "SYMBOL_SCOPE {";					\
-+	echo "    global:";					\
-+	for f in `grep _Unwind_ $< | sort`; do			\
-+	  echo "	$$f { FLAGS = EXTERN DIRECT };";	\
-+	done;							\
- 	echo "};" ) > $@
- 
-+endif
-+
- # Copy libgcc-unwind.map to the place where gcc will look for it at build-time.
- install-libgcc-unwind-map-forbuild: libgcc-unwind.map
- 	dest=$(gcc_objdir)/tmp$$$$-$<; \
--- 
-2.14.1
-
diff --git a/gcc7/gcc-7.3.0-il-1.tar.gz.sha1 b/gcc7/gcc-7.3.0-il-1.tar.gz.sha1
deleted file mode 100644
index ff52e2e..0000000
--- a/gcc7/gcc-7.3.0-il-1.tar.gz.sha1
+++ /dev/null
@@ -1 +0,0 @@
-3627ce752b9be1f0a1a86488cca6905ae942be1e
diff --git a/gcc7/gcc-gcc-7.3.0-il-3.tar.gz.sha1 b/gcc7/gcc-gcc-7.3.0-il-3.tar.gz.sha1
new file mode 100644
index 0000000..bb70756
--- /dev/null
+++ b/gcc7/gcc-gcc-7.3.0-il-3.tar.gz.sha1
@@ -0,0 +1 @@
+46e54bbe041cefd8395574d9fb686efecdc7ebce
